File: 0078-subsets/0078-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        l=[]
        for i in range(2**len(nums)):
            k=[]
            for j in range(len(nums)):
                if((i&1<<j)!=0):
                    k.append(nums[j])
            l.append(k)
        return l
        

File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>>sets;
        int r=pow(2,nums.size());
        for (int i=0;i<r;i++){
            vector<int>s;
            for(int k=0;k<nums.size();k++){
                if(i&(1<<k)){
                    s.push_back(nums[k]);
                }
            }
            sets.push_back(s);
        }
        return sets;
        
    }
};